test/instances/lseth/LsETHHyperdrive.t.sol

```diff
@@ -0,0 +1,901 @@
+        RIVER.approve(address(hyperdrive), shortAmount);
+        (, uint256 sharesPaid) = openShort(bob, shortAmount, false);
+        uint256 basePaid = RIVER.underlyingBalanceFromShares(sharesPaid);
+
+        // Ensure that the amount of base paid by the short is reasonable.
+        uint256 realizedRate = HyperdriveUtils.calculateAPRFromRealizedPrice(
+            shortAmount - basePaid,
+            shortAmount,
+            1e18
+        );
+        assertGt(basePaid, 0);
+        assertGe(realizedRate, FIXED_RATE);
+
+        // Ensure that River aggregates and the token balances were updated
+        // correctly during the trade.
+        verifyDeposit(
+            bob,
+            sharesPaid,
+            false,
+            totalLsethSupplyBefore,
+            bobBalancesBefore,
+            hyperdriveBalancesBefore
+        );
+    }
+
+    function test_open_short_refunds() external {
+        vm.startPrank(bob);
+
+        // Ensure that the refund fails when Bob sends excess ETH
+        // when opening a short with "asBase" set to true.
+        vm.expectRevert(IHyperdrive.UnsupportedToken.selector);
+        hyperdrive.openShort{ value: 2e18 }(
+            1e18,
+            1e18,
+            0,
+            IHyperdrive.Options({
+                destination: bob,
+                asBase: true,
+                extraData: new bytes(0)
+            })
+        );
+
+        // Ensure that Bob receives a refund when he opens a short with "asBase"
+        // set to false and sends ETH to the contract.
+        uint256 sharesPaid = 1e18;
+        uint256 ethBalanceBefore = address(bob).balance;
+        RIVER.approve(address(hyperdrive), sharesPaid);
+        hyperdrive.openShort(
+            sharesPaid,
+            sharesPaid,
+            0,
+            IHyperdrive.Options({
+                destination: bob,
+                asBase: false,
+                extraData: new bytes(0)
+            })
+        );
+        assertEq(address(bob).balance, ethBalanceBefore);
+    }
+
+    function test_close_short_with_eth(
+        uint256 shortAmount,
+        int256 variableRate
+    ) external {
+        // Accrue interest for a term to ensure that the share price is greater
+        // than one.
+        advanceTime(POSITION_DURATION, 0.05e18);
+
+        // Bob opens a short by depositing lsETH.
+        vm.startPrank(bob);
+        shortAmount = shortAmount.normalizeToRange(
+            100 * hyperdrive.getPoolConfig().minimumTransactionAmount,
+            HyperdriveUtils.calculateMaxShort(hyperdrive)
+        );
+        RIVER.approve(address(hyperdrive), shortAmount);
+        (uint256 maturityTime, ) = openShort(bob, shortAmount, false);
+
+        // Bob attempts to close the short after the position
+        // duration for ETH. The interest rate range is limited because
+        // the test case not fail when there is zero accrued interest.
+        variableRate = variableRate.normalizeToRange(1e18, 2.5e18);
```

Comment:
This is probably higher than it needs to be. I ran into the same edge case in another integration and was able to use `0.01e18` as the lower bound: https://github.com/delvtech/hyperdrive/blob/ee0d8cf64bef6de897bfc7532a2684ed5534d7f5/test/instances/steth/StETHHyperdrive.t.sol#L687